Emulator/PAK.Stream.Emulator/PakEmulatorApi.cs

```diff
@@ -79,4 +88,118 @@ public RouteGroupTuple[] GetEmulatorInput()
 
         return result;
     }
+
+    public ReadOnlyMemory<byte>? GetEntry(Strim pak, string entryPath)
+    {
+        var entry = ReadFileFromPak(pak, entryPath);
+        if (entry == null) return null;
+        return entry.AsMemory();
+    }
+
+    private byte[]? ReadFileFromPak(Strim fileStream, string index, string fileRoot = null)
```

Comment:
Some nullables to fix here, e.g. `fileRoot` should be `string?` (nullable)

Aside from that; isn't this code mostly duplicated with `PakEmulatorTests`?
Consider making a helper method, and using the same code in both tests and API/runtime.

Reply:
I refactored the duplicate `ReadFileFromPak`s in 2dccf6b.

I also did some (quite lazy) refactoring of `PakBuilder`'s `Build` function in 000aa1b so it's now three separate functions. I still think it is far from ideal (there's a lot of duplicate code) but I guess it's a bit more readable. I don't really want to spend any more time on it for now given it's not really got anything to do with this pr (a problem for later :D)